base/compiler/abstractinterpretation.jl

```diff
@@ -280,6 +302,17 @@ function abstract_call_gf_by_type(interp::AbstractInterpreter, @nospecialize(f),
             delete!(sv.pclimitations, caller)
         end
     end
+    # if refinement information on slot types is available, apply it now
```

Comment:
hmm I think we can't use this refinement information if we don't add backedges ..